crates/apollo-compiler/src/database/hir_db.rs

```diff
@@ -941,20 +959,34 @@ fn value(val: ast::Value) -> Value {
         }),
         ast::Value::StringValue(string_val) => Value::String(string_val.into()),
         ast::Value::FloatValue(float) => Value::Float(Float::new(float.into())),
-        ast::Value::IntValue(int) => Value::Int(int.into()),
+        ast::Value::IntValue(int) => {
+            if let Some(Type::Named { name, .. }) = expected_type {
+                if name == "Float" {
+                    return Value::Float(Float::new(int.as_float()));
```

Comment:
After #371 you can do `f64::try_from(int).unwrap()` here. (we will still panic for now, as not doing that is a bigger refactor 😬 )